test/integration/jobset_controller_test.go

```diff
@@ -311,6 +308,87 @@ var _ = ginkgo.Describe("JobSet controller", func() {
 			gomega.Eventually(checkJobSetStatus, timeout, interval).WithArguments(js, jobset.JobSetFailed).Should(gomega.Equal(true))
 		})
 	})
+	ginkgo.When("a jobset is created with suspend disabled", func() {
+		ginkgo.It("should create all jobs and headless services, then complete successfully once all jobs are completed", func() {
+			ginkgo.By("creating a new JobSet")
+			ctx := context.Background()
+
+			// Construct JobSet.
+			js := testing.MakeJobSet("suspend-false", ns.Name, false).
+				AddReplicatedJobs(testing.MakeReplicatedJob("test-job").
+					SetJob(testing.MakeJob("test-job", ns.Name).JobTemplateSpec).
+					Obj(), 3).Obj()
+
+			// Create JobSet.
+			gomega.Expect(k8sClient.Create(ctx, js)).Should(gomega.Succeed())
+
+			// We'll need to retry getting this newly created JobSet, given that creation may not immediately happen.
+			ginkgo.By("checking JobSet was created successfully")
+			gomega.Eventually(k8sClient.Get(ctx, types.NamespacedName{Name: js.Name, Namespace: js.Namespace}, &jobset.JobSet{}), timeout, interval).Should(gomega.Succeed())
+
+			ginkgo.By("checking JobSet eventually has 3 active jobs")
+			var childJobList batchv1.JobList
+			gomega.Eventually(func() (int, error) {
+				if err := k8sClient.List(ctx, &childJobList, client.InNamespace(js.Namespace)); err != nil {
+					return -1, err
+				}
+				return len(childJobList.Items), nil
+			}, timeout, interval).Should(gomega.Equal(3))
+
+			ginkgo.By("checking JobSet status is completed once all its jobs are completed")
+			// Mark jobs as complete.
+			for _, job := range childJobList.Items {
+				job.Status.Conditions = append(job.Status.Conditions, batchv1.JobCondition{
+					Type:   batchv1.JobComplete,
+					Status: corev1.ConditionTrue,
+				})
+				gomega.Expect(k8sClient.Status().Update(ctx, &job)).Should(gomega.Succeed())
+			}
+			// Check JobSet has completed.
+			gomega.Eventually(checkJobSetStatus, timeout, interval).WithArguments(js, jobset.JobSetCompleted).Should(gomega.Equal(true))
+		})
+
+	})
+	ginkgo.When("a jobset is created with suspend enabled", func() {
+		ginkgo.It("should create all jobs in a suspended state", func() {
```

Comment:
this is good to have, but lets add a test that suspends the job after it started, something like:

1. start -> all jobs should have suspend=false and the suspend condition on jobset is not existent
2. suspend -> all jobs should have suspend=true and the suspend condition on jobset is true
3. restart -> all jobs should have suspend=false and the suspend condition on jobset is false

Reply:
https://github.com/kubernetes-sigs/jobset/pull/20/files#diff-2c3dfd6ff1abd441d5beb3096d5cbf53d91218680c176d6df3549f835f28a0dcR355

I am working on this but for some reason I am unable to update a JobSet once it is set in the integration tests.  

Do you have an idea of why this would be the case?  

I'm getting:

```
     Operation cannot be fulfilled on jobsets.batch.x-k8s.io "suspend-true": 
the object has been modified
; please apply your changes to the latest version and try again
```

Reply:
I think this is because the status was updated, so you need to get it again before you do the second update.

Reply:
Yep! Found that out and I think it’s ready for another review.